Documentation/ch-scrub-intro.rst

```diff
@@ -6,14 +6,27 @@ On filesystems that use replicated block group profiles (e.g. raid1), scrub will
 also automatically repair any damage by default by copying verified good data
 from one of the other replicas.
 
+.. warning::
+   Files with the NOCOW (``chattr +C``) attribute imply ``nodatasum``.
+   Errors in these files cannot be detected or corrected by scrub. This means
+   the NOCOW attribute is not currently safe in the presence of replicas since
+   btrfs cannot determine which copy of data to use after a failure and may
+   return data from any replica, including corrupt or outdated data.
+   Detecting and recovering from a failure in this case requires manual
+   intervention. See issue `#482 <https://github.com/kdave/btrfs-progs/issues/482>`_.
+   
+   Notably, `systemd sets NOCOW on journals by default <https://github.com/systemd/systemd/commit/11689d2a021d95a8447d938180e0962cd9439763>`_,
+   and `libvirt ≥ 6.6 sets NOCOW on storage pool directories by default <https://www.libvirt.org/news.html#v6-6-0-2020-08-02>`_.
+   Other distributions may also enable NOCOW on database files or directories to
```

Comment:
And it's not only users' choice, but certain IO pattern can lead to csum mismatch by themselves (although that counts as an abuse though).

Reply:
I feel like this "feature" is a huge failure on the part of the libvirt developers. Checksums are super important for data integrity and they are disabled by this "default behaviour". For me to only find out about it after 4 years thanks to this PR feels embarrassing.

I checked all my servers that use btrfs and found that most did not have this problem - but that is likely because the image subvolumes and files were created via automation before the storage pool configurations were created. However, whenever manual action was taken, there was always a chance that the storage pool configuration would create the path with nocow.